### PR TITLE
Fixed issue with type deduced expressions breaking ternary operators

### PR DESCRIPTION
**Pull request by thorulf4**

### CODE REVIEW

File: src/typechecker.cpp [hunk 1]
```diff
@@ -2083,11 +2097,12 @@ bool TypeChecker::checkExpression(expression_t expr)
             handleError(expr, "$First_argument_of_inline_if_must_be_an_integer");
             return false;
         }
-        if (!areInlineIfCompatible(expr.get_type(), expr[1].get_type(), expr[2].get_type())) {
+
+        type = getInlineIfCommonType(expr[1].get_type(), expr[2].get_type());
+        if (!areInlineIfCompatible(type, expr[1].get_type(), expr[2].get_type())) {
```
Comment (mikucionisaau) [inline src/typechecker.cpp:2102]: It should be possible to convert the common type to the resulting type
```suggestion
        if (!areAssignmentCompatible(resultType, type)) {
```

Comment (thorulf4) [reply src/typechecker.cpp:2106]: I'm not sure what `resultType` is supposed to be. But `areInlineIfCompatible` call does `areAssignmentCompatible` checks

**Comment by thorulf4** [comment]
I've run regression and uppaal tests, both succeed

File: src/typechecker.cpp (hunk 1 above)
Comment (mikucionisaau) [reply src/typechecker.cpp:2106]: given the expression `res = (c) ? (a) : (b)` then the `resultType` is the type of `res`.

**Review by mikucionisaau — APPROVED** [review]

**Pull request merged**